src/game.cpp

```diff
@@ -10517,22 +10531,22 @@ bool game::walk_move( const tripoint &dest_loc, const bool via_ramp, const bool
     }
     u.set_underwater( false );
 
-    if( !shifting_furniture && !pushing && is_dangerous_tile( dest_loc ) ) {
-        std::vector<std::string> harmful_stuff = get_dangerous_tile( dest_loc );
+    std::vector<std::string> harmful_stuff = get_dangerous_tile( dest_loc );
+    if( !shifting_furniture && !pushing && !harmful_stuff.empty() ) {
```

Comment:
If the tile is dangerous, we have to run `get_` for the full list.
If the tile isn't dangerous, then `is_` is going to perform the whole search before bailing out.

So, either way, we have to do the full search, and might as well do it just once.